classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -32,752 +34,1148 @@ abstract protected function getContextTable(): string;
+                    )
+                )
+                ->whereNull('p.publication_id')
+                ->pluck('s.submission_id', 'last.publication_id');
+            foreach ($rows as $publicationId => $submissionId) {
+                $this->_installer->log("The current publication ID ({$publicationId}) for the submission ID {$submissionId} is invalid, the publication ID {$publicationId} will replace it");
+                $affectedRows += DB::table('submissions')->where('submission_id', '=', $submissionId)->update(['current_publication_id' => $publicationId]);
             }
 
+            // The current_publication_id is nullable, but it's in fact required by the software, so we delete orphan entries instead of nulling them
+            $affectedRows += $this->deleteRequiredReference('submissions', 'current_publication_id', 'publications', 'publication_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('submission_files', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~5 entities: file_id->files.file_id genre_id->genres.genre_id source_submission_file_id->submission_files.submission_file_id submission_id->submissions.submission_id uploader_user_id->users.user_id
+            $affectedRows += $this->deleteRequiredReference('submission_files', 'submission_id', 'submissions', 'submission_id');
+            $affectedRows += $this->deleteRequiredReference('submission_files', 'file_id', 'files', 'file_id');
+            $affectedRows += $this->cleanOptionalReference('submission_files', 'uploader_user_id', 'users', 'user_id');
+            $affectedRows += $this->cleanOptionalReference('submission_files', 'source_submission_file_id', 'submission_files', 'submission_file_id');
+            $affectedRows += $this->cleanOptionalReference('submission_files', 'genre_id', 'genres', 'genre_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('publications', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~4 entities: primary_contact_id->authors.author_id doi_id->dois.doi_id(not found in previous version) section_id->sections.section_id submission_id->submissions.submission_id
+            $affectedRows += $this->deleteRequiredReference('publications', 'submission_id', 'submissions', 'submission_id');
+            $affectedRows += $this->cleanOptionalReference('publications', 'primary_contact_id', 'authors', 'author_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('user_groups', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~1 entities: context_id->context_table.context_id
+            // Custom field (not found in at least one of the softwares)
+            $affectedRows += $this->cleanOptionalReference('user_groups', 'context_id', $this->getContextTable(), $this->getContextKeyField(), $this->ignoreZero('context_id'));
```

Comment:
I don't think this should be an optional reference. There is a missing entry in `journals` in the SciELO data set, and I get a fatal error here where it attempts to set `context_id` to `null` instead of deleting those entries:
```
ERROR: Upgrade failed: DB: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'context_id' cannot be null (SQL: update `user_groups` set `context_id` = ? where `context_id` in (180))
```

Reply:
Here's the reason for the `null` `context_id`
![image](https://github.com/pkp/pkp-lib/assets/361921/97be99b1-41e8-4d13-bc46-a2db1ed627ad)

Reply:
Yes, the `context_id` will be `0` for the site administrator role, and that should become a `null` for the sake of adding a FK constraint. But let's back off adding these constraints for 3.4.0 (as [noted here](https://github.com/pkp/pkp-lib/issues/8929#issuecomment-1567495168)) to avoid a regression for 3.4.0.